tests/mock_s3_server/mock_s3_server.py

```diff
@@ -275,11 +300,19 @@ def handle_get_object_modified(start_range, end_range, request):
         with open(response_file, 'r') as f:
             data = json.load(f)
             if data['headers']['ETag'] == etag:
-                return "/get_object_modified_success", data_length, False
-            return "/get_object_modified_failure", data_length, False
+                return Response("/get_object_modified_success")
+            return Response("/get_object_modified_failure")
+
 
+def handle_get_object(wrapper, request, parsed_path, head_request=False):
+    wrapper.info("handle_get_object")
+    if parsed_path.path == "/get_object_checksum_retry" and not head_request:
+        TrioHTTPWrapper.retry_request_received_continuous = TrioHTTPWrapper.retry_request_received_continuous + 1
 
-def handle_get_object(request, parsed_path):
+        wrapper.info("retry_request_received_continuous is " +
+                     str(TrioHTTPWrapper.retry_request_received_continuous))
+    else:
+        TrioHTTPWrapper.retry_request_received_continuous = 0
```

Comment:
I was trying to suggest a way to use the Response object to communicate between the 2 functions, rather than relying on 1 function setting a singleton that another function checks. I realize you'll need a singleton to count retries, but hoped we use keep its use local to 1 function

Like, you access the singleton here, and use that to set response.fail_after_sending_headers

then in the send_response_from_json() function you do `if response.fail_after_sending_headers:` instead of `if TrioHTTPWrapper.retry_request_received_continuous == 1`